upup/pkg/fi/cloudup/awstasks/autoscalinglifecyclehook.go

```diff
@@ -47,11 +47,20 @@ type AutoscalingLifecycleHook struct {
 }
 
 var _ fi.CompareWithID = &AutoscalingLifecycleHook{}
+var _ fi.CloudupHasDependencies = &AutoscalingLifecycleHook{}
 
 func (h *AutoscalingLifecycleHook) CompareWithID() *string {
 	return h.Name
 }
 
+func (e *AutoscalingLifecycleHook) GetDependencies(tasks map[string]fi.CloudupTask) []fi.CloudupTask {
```

Comment:
Probably needed due to https://github.com/kubernetes/kops/pull/15722. Any thoughts?

Reply:
#15722 required an explicit GetDependencies because the reflection-based implementation provided an unwanted dependency. That situation doesn't apply here.

Reply:
I understand what you are saying but seems that adding that fix may affect things depending on ASGs.
As you can see, I only left this in the PR and it seems to fix the issue.